lib/ReactViews/Notification/MapInteractionWindow.tsx

```diff
@@ -0,0 +1,148 @@
+"use strict";
+
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+const parseCustomHtmlToReact = require("../Custom/parseCustomHtmlToReact");
+import Styles from "./map-interaction-window.scss";
+import classNames from "classnames";
+import { observer } from "mobx-react";
+import MapInteractionMode, { UIMode } from "../../Models/MapInteractionMode";
+import { observable, Lambda, reaction } from "mobx";
+import isDefined from "../../Core/isDefined";
+import Terria from "../../Models/Terria";
+import ViewState from "../../ReactViewModels/ViewState";
+
+const MapInteractionWindowWrapper = styled.div<{ isDiffTool: boolean }>`
+  ${props =>
+    props.isDiffTool &&
+    `
+    top: initial;
+    bottom: 100px;
+    min-width: 330px;
+    width: auto;
+
+    box-sizing: border-box;
+    padding: 10px 15px;
+    background: ${props.theme.colorSplitter};
+    color:${props.theme.textLight};
+  `}
+`;
+
+@observer
+export default class MapInteractionWindow extends React.Component<{
+  terria: Terria;
+  viewState: ViewState;
+}> {
+  static propTypes = {
+    terria: PropTypes.object,
+    viewState: PropTypes.object
+  };
+
```

Comment:
I don't know react well, do you need `propTypes` and React.Component generic props definition?